pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,532 @@
+
+            Calculate the expected number of repeat purchases up to time t for a
+            randomly choose individual from the population, given they have
+            purchase history (frequency, recency, T).
+
+            See equation (41) from [2]_.
+
+            Adapted from lifetimes package
+        https://github.com/CamDavidsonPilon/lifetimes/blob/41e394923ad72b17b5da93e88cfabab43f51abe2/lifetimes/fitters/pareto_nbd_fitter.py#L242
+            Parameters
+            ----------
+            future_t: array_like
+                times to calculate the expectation for.
+            customer_id: array_like
+                Customer labels.
+            recency: array_like
+                Number of time periods between the customer's first and most recent purchases.
+            frequency: array_like
+                Number of repeat purchases per customer.
+            T: array_like
+                Number of time periods since the customer's first purchase.
+                Model assumptions require T >= recency.
+        """
+
+        x, t_x, T = self._process_customers(customer_id, frequency, recency, T)
+        r, alpha, s, beta = self._unload_params()
+        loglike = self._logp(r, alpha, s, beta, x, t_x, T)
+
+        first_term = (
+            gammaln(r + x)
+            - gammaln(r)
+            + r * log(alpha)
+            + s * log(beta)
+            - (r + x) * log(alpha + T)
+            - s * log(beta + T)
+        )
+        second_term = log(r + x) + log(beta + T) - log(alpha + T)
+        third_term = log(
+            (1 - ((beta + T) / (beta + T + future_t)) ** (s - 1)) / (s - 1)
+        )
+
+        exp_purchases = exp(first_term + second_term + third_term - loglike)
+
+        return exp_purchases.transpose(
+            "chain", "draw", "customer_id", missing_dims="ignore"
+        )
+
+    def expected_purchases_new_customer(
+        self,
+        t: Union[np.ndarray, pd.Series],
+    ) -> xarray.DataArray:
+        r"""
+            Expected number of purchases for a new customer across :math:`t` time periods. See
+            equation (27) of [2]_.
+
+            http://brucehardie.com/notes/009/pareto_nbd_derivations_2005-11-05.pdf
+
+            Adapted from lifetimes package
+        https://github.com/CamDavidsonPilon/lifetimes/blob/41e394923ad72b17b5da93e88cfabab43f51abe2/lifetimes/fitters/pareto_nbd_fitter.py#L359
+
+            Parameters
+            ----------
+            t: array_like
+                Number of time periods over which to estimate purchases.
+        """
+        r, alpha, s, beta = self._unload_params()
+
+        t = np.asarray(t)
+
+        r, alpha, s, beta = self._unload_params()  # type: ignore [has-type]
+        first_term = r * beta / alpha / (s - 1)
+        second_term = 1 - (beta / (beta + t)) ** (s - 1)
+
+        return (first_term * second_term).transpose(
+            "chain", "draw", "t", missing_dims="ignore"
+        )
+
+    def probability_alive(
```

Comment:
I have been trying to be consistent in distinguishing  between expectations and full distributions, by using expected/distribution as a prefix. It's a bit verbose but I think more clear and more important for us than for lifetimes since we are working with posterior distributions by default.

```suggestion
    def expected_probability_alive(
```